1-Draft/RFCXXXX-Umask-and-Ulimit.md

```diff
@@ -0,0 +1,276 @@
+
+### Two New Cmdlets
+
+The PowerShell equivalent shall break these into two separate cmdlets:
+
+* Get-Ulimit
+* Set-Ulimit
+
+### Get-Ulimit
+
+Get
+
+#### Get-Ulimit Syntax
+
+```powershell
+NAME
+    Get-Ulimit
+
+SYNTAX
+    Get-Ulimit [[-Resource] {CpuTime | FileSize | DataSegmentSize | StackSize | CoreFileSize | AddresSpaceSize | MemoryAddressSize | 
+    NumberOfProcesses | NumberOfOpenFiles}] [<CommonParameters>]
+
+    Get-Ulimit [-All] [<CommonParameters>]
+
+```
+
+#### Get-Ulimit Examples
+
+```powershell
+
+PS> get-ulimit
+
+Resource             Current             Maximum
+--------             -------             -------
+FileSize 9223372036854775807 9223372036854775807
+
+PS> get-ulimit -Resource FileSize
+
+Resource             Current             Maximum
+--------             -------             -------
+FileSize 9223372036854775807 9223372036854775807
+
+PS> get-ulimit -resource FileSize,NumberOfOpenFiles,NumberOfProcesses
+
+         Resource             Current             Maximum
+         --------             -------             -------
+         FileSize 9223372036854775807 9223372036854775807
+NumberOfOpenFiles               10240 9223372036854775807
+NumberOfProcesses                2837                4256
+
+PS> get-ulimit -all
+
+         Resource             Current             Maximum
+         --------             -------             -------
+          CpuTime 9223372036854775807 9223372036854775807
+         FileSize 9223372036854775807 9223372036854775807
+  DataSegmentSize 9223372036854775807 9223372036854775807
+        StackSize             8388608            67104768
+     CoreFileSize                   0 9223372036854775807
+  AddresSpaceSize 9223372036854775807 9223372036854775807
+MemoryAddressSize 9223372036854775807 9223372036854775807
+NumberOfProcesses                2837                4256
+NumberOfOpenFiles               10240 9223372036854775807
+
+
+PS> get-ulimit -all |get-member
+
+   TypeName: System.Management.Automation.Platform+ResourceLimitInfo
+Name        MemberType Definition
+----        ---------- ----------
+Equals      Method     bool Equals(System.Object obj)
+GetHashCode Method     int GetHashCode()
+GetType     Method     type GetType()
+ToString    Method     string ToString()
+Current     Property   ulong Current {get;set;}
+Maximum     Property   ulong Maximum {get;set;}
+Resource    Property   System.Management.Automation.Platform+ResourceForUlimit Resource {get;set;}
+```
+
+#### Implementation Notes
+
+The `ulimit` cmdlets do not totally replicate the Unix behavior.
+Specifically, resources which are not alterable via the `setrlimit` API will not be supported.
+The pipe size resource is not retrievable or alterable with the ulimit cmdlets.
+The resources available will vary based on the platform, the following table shows what resources shall be returned based on their platform:
```

Comment:
On Windows we have Windows System Resource Manager and we could create ported cmdlet.

Reply:
Given the UNIX-specific nature of ulimit, it might be best to have this as the underlying primitive rather than try to balance multiple APIs